uui/components/overlays/__tests__/Modals.test.tsx

```diff
@@ -165,5 +166,155 @@ describe('Modals', () => {
         expect(screen.getByText('Test content')).toBeInTheDocument();
     });
 
+    it('should properly handle closing picker input\'s options and modal window on \'Escape\' key presses', async () => {
+        const successMock = jest.fn();
+        const abortMock = jest.fn();
+        const { wrapper, testUuiCtx } = getDefaultUUiContextWrapper();
+
+        function ModalWindowWithPickerInput(
+            props: IModal<undefined>,
+        ) {
+            interface Option {
+                id: string;
+                name: string;
+            }
+
+            const [
+                option,
+                setOption,
+            ] = useState<Option | undefined>(undefined);
```

Comment:
Initially, I just passed `undefined` and an empty function to `value` and `onValueChange`, but decided to make the implementation as close as possible to how `PickerInput` is used in real world scenarios.